src/ru/hotmanga/src/eu/kanade/tachiyomi/extension/ru/hotmanga/HotManga.kt

```diff
@@ -0,0 +1,327 @@
+package eu.kanade.tachiyomi.extension.ru.hotmanga
+
+import android.app.Application
+import android.content.SharedPreferences
+import android.util.Log
+import android.widget.Toast
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.extension.ru.hotmanga.dto.MangaDto
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromJsonElement
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import rx.Observable
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import java.net.URI
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.concurrent.TimeUnit
+
+class HotManga : ConfigurableSource, HttpSource() {
+
+    override val id = 2073023199372375753
+
+    private val preferences: SharedPreferences by lazy {
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
+    }
+
+    private val baseOrig: String = "https://hotmanga.me"
+
+    private val baseMirr: String = "https://xn--80aaaklzpjd4c4a.xn--p1ai" // https://мангахентай.рф
+
+    private val baseMirrSecond: String = "https://xn--80aaalhzvfe9b4a.xn--80asehdb" // https://хентайманга.онлайн
+
+    private val baseMirrThird: String = "https://xn--80aanrbklcdf5b7a.xn--p1ai" // https://хентайонлайн.рф
+
+    private val apiPath = "/api"
+
+    private val domain: String? = preferences.getString(DOMAIN_PREF, baseOrig)
+
+    private val paidSymbol = "\uD83D\uDCB2"
+
+    override val baseUrl = domain.toString()
+
+    override val lang = "ru"
+
+    override val name = "HotManga"
+
+    override val supportsLatest = true
+
+    private val apiPathsMap = mapOf(
+        baseOrig to apiPath,
+        baseMirr to "/api-frontend",
+        baseMirrSecond to apiPath,
+        baseMirrThird to apiPath,
+    )
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimit(3)
+        .connectTimeout(5, TimeUnit.MINUTES)
+        .readTimeout(30, TimeUnit.SECONDS)
+        .writeTimeout(15, TimeUnit.SECONDS).build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Referer", baseUrl)
```

Comment:
Referer header has a trailing slash.
```suggestion
        .set("Referer", "$baseUrl/")
```